enhancements/SEP1-enhancement-process.md

```diff
@@ -0,0 +1,53 @@
+|Status                                             | Authors      | Created    | 
+|---------------------------------------------------|--------------|------------|
+|WIP                                                | @dgn         | 2024-07-09 |
+
+# Sail Enhancement Proposal (SEP) Process
+
+## Overview
+
+
+## Goals
+For non-trivial (for definition see below) changes to the sail-operator, we're going to use an open design process that allows us to:
+1. discuss designs and potential APIs in detail where GitHub issues might not be sufficient,
+1. record important architectural decisions so we can find and reference them later on,
+1. identify functional and non-functional requirements before implementation,
+1. conduct our design in the open so that community members can participate and
+1. use asynchronous communication throughout to minimize meeting load.
+
+## Non-goals
+1. Overcomplicating the engineering process. Simple things should stay simple - just submit a PR to fix that bug. It's the hard things we want to make sure to get right using this process. But that doesn't mean you have to write pages of prose. Keep the SEPs short and concise, it will make reviews easier and speed up the process.
+
+## Design
+
+### Applicability
+The SEP process should be used for non-trivial features or epics and any API changes. Changes without user impact will most of the time not need a SEP, but if they're big enough or result in changes to the development workflows (e.g. a rewrite of our integration testing framework) it's probably a good idea to create one.
```

Comment:
Work related to an enhancement can sometimes span across multiple repositories:

- Entirely within sail-operator.
- Partially in sail-operator and partially in core Istio (as an example).
- Completely outside of sail-operator.

Should we clarify when a user is expected to use the SEP?

Reply:
Good point. I'd argue that the SEP can include potentially needed work in Istio, but they might require yet another design doc if the changeset is big. We would need to link the docs in that case

Reply:
I added a paragraph

Reply:
Thanks @dgn. LGTM, but I'll let others to take a look.